### PR TITLE
feat_: graceful shutdown with status-backend

### LINKED ISSUE #6306: feat_: graceful shutdown with status-backend

### PR DESCRIPTION
**Pull request by igor-sirotin**
1. This is just common sense to do
2. It's required for functional tests, because coverage data is only saved correctly with graceful shutdown of the app
3. Added support for `SIGTERM`
For `statusd` we only support `SIGINT`, so I had to change the emitted signal. We won't need to do it for `status-backend`.
https://github.com/status-im/status-go/blob/35dc84fa7fb56b7343accf0e1381a47946a27057/tests-functional/docker-compose.test.status-go.yml#L28

closes https://github.com/status-im/status-go/issues/6306

### CODE REVIEW

File: cmd/status-backend/main.go [hunk 1]
```diff
@@ -57,3 +61,17 @@ func main() {
 	srv.RegisterMobileAPI()
 	srv.Serve()
 }
+
+// haltOnInterruptSignal catches interrupt signal (SIGINT) and
+// stops the node. It times out after 5 seconds
```
Comment (osmaczko) [inline cmd/status-backend/main.go]: Where is this 5 seconds timeout defined? :thinking:

**Review by osmaczko — APPROVED** [review]

File: cmd/status-backend/main.go (hunk 1 above)
Comment (igor-sirotin) [reply cmd/status-backend/main.go]: oops, copy pasted from `statusd`
I actually read this and went to check if there's a timeout somewhere, but forgot to remove 🤦

**Review by antdanchenko — APPROVED** [review]

File: cmd/status-backend/main.go (hunk 1 above)
Comment (osmaczko) [reply cmd/status-backend/main.go]: Timeout was a good thing, no?

Comment (igor-sirotin) [reply cmd/status-backend/main.go]: @osmaczko I think the timeout should be defined by externally. One could always force quit the process without expecting it for graceful shutdown. And I'm not sure what would be the timeout, how would we define it 🤔 

And anyway, the comment was misliading, there's no timeout for this function `statusd`

Comment (osmaczko) [reply cmd/status-backend/main.go]: Yeah, I agree, good point. It should be up to tests to send SIGKILL on timeout.

**Pull request merged**